### PR TITLE
added unspecified endpoints warning (#88)

### PR DESCRIPTION
**Pull request by YairSlobodin1**

### CODE REVIEW

**Comment by kyorav** [comment]
@YairSlobodin1 - testing that a warning was added to the log is complicated. Lets create a separate issue for that.

File: pkg/ir/unspecified.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+		if !subnetFound {
+			warningString, printWarning = s.updateWarningString(printWarning, warningString, subnet)
+		}
+	}
+	if printWarning {
+		log.Println(warningString + ".")
+	}
+}
+
+func (s *Spec) findEndPointsInConnections(endpoints []string, warningString string, printWarning bool) (string, bool) {
```
Comment (zivnevo) [inline pkg/ir/unspecified.go:72]: I believe it's better if this function will receive just a slice of endpoints (as strings), and return a slice with those endpoints that have no connections.
Then you can later append to the returned slice more endpoints if needed, decide to print or not based on its size, and join all the endpoints to one string using `strings.Join()`

File: pkg/ir/unspecified.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+)
+
+func (s *Spec) ComputeBlockedEndPoints() {
+	printWarning := false
+	warningString := "The following endpoints do not have required connections; no SGs were generated for them: "
+
+	instances := make([]string, 0, len(s.Defs.InstanceToNIFs))
+	for instance := range s.Defs.InstanceToNIFs {
+		instances = append(instances, instance)
+	}
```
Comment (zivnevo) [inline pkg/ir/unspecified.go:14]: In the `utils` package Elazar added a function `SortedKeys()` that takes a map and returns all its keys

File: pkg/ir/unspecified.go [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+func (s *Spec) findSubnetInConnections(subnet string) bool {
+	for c := range s.Connections {
+		if s.Connections[c].Src.Type == EndpointTypeSubnet && subnet == s.Connections[c].Src.Name {
+			return true
+		}
+		if s.Connections[c].Dst.Type == EndpointTypeSubnet && subnet == s.Connections[c].Dst.Name {
+			return true
+		}
+	}
+	return false
```
[the remark this answers is not in the document]
Comment (YairSlobodin1) [reply pkg/ir/unspecified.go:139]: When we look for the subnet in spec, we should check whether the endpoint type is `EndpointTypeSubnet`
for segments, we should check whether the endpoint type is `EndpointTypeSegment`. However, segments are saved as `EndpointTypeSubnet` in spec. I opened an issue about it (#101).

**Review by zivnevo — APPROVED** [review]
LGTM

**Pull request merged**